presto-orc/src/test/java/com/facebook/presto/orc/IntegerDictionaryProviderTest.java

```diff
@@ -0,0 +1,302 @@
+        for (Map.Entry<NodeId, long[]> entry : dictionaryStreams.entrySet()) {
+            StreamId streamId = entry.getKey().toDictionaryDataStreamId();
+            long[] data = entry.getValue();
+            long[] dictionary = new long[0];
+            long[] newDictionary = dictionaryProvider.getDictionary(createFlatStreamDescriptor(streamId), dictionary, data.length);
+            assertNotEquals(newDictionary, dictionary);
+            assertEquals(newDictionary, data);
+        }
+    }
+
+    @Test(dataProvider = "dataForTestDataCorruptionCheck", expectedExceptions = OrcCorruptionException.class, expectedExceptionsMessageRegExp = ".* Dictionary is not empty but data stream is not present.*")
+    public void testDataCorruptionCheck(Map<NodeId, long[]> dictionaryStreams, NodeId missingNode)
+            throws Exception
+    {
+        IntegerDictionaryProvider dictionaryProvider = new IntegerDictionaryProvider(createLongDictionaryStreamSources(dictionaryStreams));
+
+        StreamId streamId = missingNode.toDictionaryDataStreamId();
+        long[] dictionary = new long[0];
+        dictionaryProvider.getDictionary(createFlatStreamDescriptor(streamId), dictionary, 0);
+    }
+
+    public Object[][] dataForDictionaryLoadingTests()
+    {
+        return new Object[][] {
+                {ImmutableMap.of(
+                        new NodeId(1, 0), new long[] {1, 2, 3, 4}),
+                        ImmutableList.of(new NodeId(0, 0),
+                                new NodeId(2, 0),
+                                new NodeId(1, 1),
+                                new NodeId(42, 0))
+                },
+                {ImmutableMap.of(
+                        new NodeId(1, 0), new long[] {1, 2, 3, 4},
+                        new NodeId(3, 0), new long[] {1, 3, 5, 7}),
+                        ImmutableList.of(new NodeId(0, 0),
+                                new NodeId(2, 0),
+                                new NodeId(1, 1),
+                                new NodeId(3, 1),
+                                new NodeId(42, 0))
+                },
+                {ImmutableMap.of(
+                        new NodeId(1, 0), new long[] {1, 2, 3, 4},
+                        new NodeId(3, 0), new long[] {1, 3, 5, 7},
+                        new NodeId(4, 1), new long[] {1, 1, 2, 3},
+                        new NodeId(4, 2), new long[] {2, 4, 6, 8},
+                        new NodeId(4, 4), new long[] {1, 4, 9, 16}),
+                        ImmutableList.of(new NodeId(2, 0),
+                                new NodeId(4, 3),
+                                new NodeId(4, 42),
+                                new NodeId(42, 0))
+                },
+        };
+    }
+
+    @DataProvider(name = "dataForDictionaryLoadingTest")
+    public Object[][] dataForDictionaryLoadingTest()
+    {
+        return dataForDictionaryLoadingTests();
+    }
+
+    private Object[][] flattenDataCorruptionCheckTestInput(Object[][] testCases)
```

Comment:
This seems like an overkill. Can we just add a simple corrupted version of data from dataForDictionaryLoadingTests?

Reply:
I wanted to check that for each configuration, what streams exist and what don't. An alternative could be to expect throw on each non-existing case and have a simplified test case for corruption for exception message.